packages/esm-patient-labs-app/src/test-results/filter/filter-set.tsx

```diff
@@ -42,8 +42,7 @@ const FilterSet: React.FC<FilterSetProps> = ({ hideFilterSetHeader = false }) =>
   const [showSearchInput, setShowSearchInput] = useState(false);
 
   const handleInputChange = useCallback(
-    (e) => {
-      setSearchTerm(searchTerm);
+    (searchTerm) => {
```

Comment:
What is the reason for this change?
Changing the event to a string will not work, until you make change in the said `onClick` function.

Reply:
@vasharma05 should I revert it to the original? Also help explain abit more as I haven't understood what you mean by the "the said onClick function". But this was my thought process: When the such term changes, the handleInputChange function is called with the search term passed as an argument. We then use that search term to get the filtered results. Previously 'e' was not being used anywhere in the handleInputChange function and I didn't understand why we are using an event as a parameter yet handleInputChange is being passed a string. Also though that maybe setting the search term( setSearchTerm(searchTerm) ) to the same value is kinda redudant. Thats how I thought about it. I am willing to be corrected

Reply:
Yeah I think @vasharma05 mistakenly thought this is an input callback. It is actually being used in a `useEffect` and passed the `searchTerm` explicitly. But so this is all sort of nonsense. The code should probably just read 

```tsx
  useEffect(() => {
      const filteredData = roots.filter((node) => filterTreeNode(searchTerm, node));
      setTreeDataFiltered(filteredData);
  }, [searchTerm, root]);
```

The `handleInputChange` function can be gotten rid of entirely.